src/nvim/eval.c

```diff
@@ -8433,11 +8433,13 @@ static void filter_map(typval_T *argvars, typval_T *rettv, int map)
   if (argvars[0].v_type == VAR_LIST) {
     if ((l = argvars[0].vval.v_list) == NULL
         || (!map && tv_check_lock(l->lv_lock, arg_errmsg, TV_TRANSLATE))) {
+      tv_copy(&argvars[0], rettv);
       return;
     }
   } else if (argvars[0].v_type == VAR_DICT) {
     if ((d = argvars[0].vval.v_dict) == NULL
         || (!map && tv_check_lock(d->dv_lock, arg_errmsg, TV_TRANSLATE))) {
+      tv_copy(&argvars[0], rettv);
```

Comment:
Here you have three tv_copy() calls like if that needs to be called before return for some reason; I meant that in the surrounding if(), but with the bottommost one removed: like if saying that first argument is always returned if it is valid.